### PR TITLE
Add computer use demo

### PR DESCRIPTION
**Pull request by x5a**

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request: {}
+  push:
+    branches:
+      - main
+jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working_directory: computer-use-demo
```
Comment (nsmccandlish) [inline .github/workflows/tests.yaml]:
```suggestion
        working-directory: computer-use-demo
```

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: astral-sh/ruff-action@v1
+  pyright:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working_directory: computer-use-demo
```
Comment (nsmccandlish) [inline .github/workflows/tests.yaml]:
```suggestion
        working-directory: computer-use-demo
```

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          python -m venv .venv
+          source .venv/bin/activate
+          pip install -r dev-requirements.txt
+      - run: echo "$PWD/.venv/bin" >> $GITHUB_PATH
+      - uses: jakebailey/pyright-action@v1
+  pytest:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working_directory: computer-use-demo
```
Comment (nsmccandlish) [inline .github/workflows/tests.yaml]:
```suggestion
        working-directory: computer-use-demo
```

**Review by nsmccandlish — APPROVED** [review]

**Review by alexalbertt — APPROVED** [review]

**Review by praboud-ant — APPROVED** [review]

File: computer-use-demo/computer_use_demo/streamlit.py [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+WARNING_TEXT = "⚠️ Security Alert: Never provide access to sensitive accounts or data, as malicious web content can hijack Claude's behavior"
+
+
+class Sender(StrEnum):
+    USER = "user"
+    BOT = "assistant"
+    TOOL = "tool"
+
+
+def setup_state():
```
Comment (zachschillaci27) [inline computer-use-demo/computer_use_demo/streamlit.py:56]: Do all these variables need to be defined in the session state? The state should be reserved for variables which must be preserved across reruns (e.g. `messages`). It's not typically useful for values which are just used within a run (e.g. `hide_images`).

**Pull request merged**